packages/web-app-files/src/spaces/views/Spaces.vue

```diff
@@ -0,0 +1,107 @@
+<template>
+  <div class="oc-p-s">
+    <h2 class="" v-text="$gettext('Spaces')"></h2>
+    <div class="">
+      <span v-text="$gettext('Access all project related files in one place.')"></span>
+      <a href="#" v-text="$gettext('Learn more about spaces.')"></a>
+      <h3 v-text="$gettext('Your spaces')"></h3>
+      <hr class="uk-margin-bottom" />
+    </div>
+    <div class="spaces-list uk-container uk-container-expand uk-padding-remove">
+      <list-loader v-if="loadSpacesTask.isRunning" />
+      <template v-else>
+        <div
+          v-if="spaces.length"
+          class="
+            uk-grid-match uk-grid-column-small uk-grid-row-large uk-text-center uk-child-width-1-3@s
+          "
+          uk-grid
+        >
+          <a v-for="space in spaces" :key="space.id" href="#">
+            <span class="spaces-list-card oc-border uk-card uk-card-default">
+              <span class="uk-card-media-top oc-border-b">
+                <img
+                  :src="space.image || defaultImg"
+                  :class="{ 'spaces-list-default-img': !space.image }"
+                  :alt="$gettext('Space image')"
+                />
+              </span>
+              <span class="uk-card-body">
+                <h3 class="uk-card-title">{{ space.name }}</h3>
+              </span>
+            </span>
+          </a>
+        </div>
+        <div v-else>
```

Comment:
The `no-content-message` component doesn't need a wrapping `div`.
Also, the other views have the following structure:
```
list-load v-if...
template v-else
   no-content-message v-if=empty
   whatever-you-want-to-display
```
Would be good if you keep the same structure, i.e. move the `no-content-message` above the actual spaces overview please.

Reply:
But then the loading spinner would overlay the whole page, not just `Your spaces`. Wouldn't it be nice to always show the header section with `Access all project related spaces...`? Same goes for the no-content-message.